csv-generator/src/main/scala/models/generator/csv/CsvGenerator.scala

```diff
@@ -0,0 +1,52 @@
+package models.generator.csv
+
+import io.apibuilder.generator.v0.models.{File, InvocationForm}
+import io.apibuilder.spec.v0.models.{Operation, Resource, Service}
+import lib.generator.CodeGenerator
+
+class CsvGenerator extends CodeGenerator {
+
+  val SEPARATOR = ","
+  val NEWLINE = "\n"
+  val HEADER = s"method${SEPARATOR}path${NEWLINE}"
+  val EMPTY = ""
+
+  override def invoke(form: InvocationForm): Either[Seq[String], Seq[File]] = {
+    Right(generateSourceFiles(form.service))
+  }
+
+  def generateSourceFiles(service: Service): Seq[File] = {
+    Seq(generateResourcesFile(service.resources))
+  }
+
+  def generateResourcesFile(resources: Seq[Resource]): File = {
+
+    val lines: Seq[Seq[String]] = for {
+      resource <- resources
+      operation <- resource.operations
+    } yield makeLine(resource, operation)
+
+    File(
+      name = "resources.csv",
+      contents = HEADER + lines.map(_.mkString(SEPARATOR)).mkString(NEWLINE)
+    )
+  }
+
+  def wrapInQuotes(input: String): String = {
+    if (input.contains(SEPARATOR))
+      "\"" + input + "\""
```

Comment:
what if input already contains a quote?

Reply:
@benwaffle description could potentially contain a quote.  But really - i'm future-proofing this.

Reply:
Sorry, I wasn't clear. I agree that this part is needed. But let's say the input already has a quote, e.g. `hello, "world"`. I believe your code would output `"hello, "world""`. That would cause problems. The way to solve it (by rfc4180) is to replace each double quote (`"`) inside the string with two quotes (`""`), resulting in `"hello, ""world"""`

Reply:
Right, now I understand your point.  Thank you.   I will update to handle quotes.

Reply:
I would use Apache [commons-csv] to handle CSV output

https://commons.apache.org/proper/commons-csv/

Reply:
https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVPrinter.html

Reply:
yeah, that's a good idea, i'll try to plug it in.

Reply:
@sullis @benwaffle - it now uses commons-csv